base/abstractarray.jl

```diff
@@ -476,7 +480,8 @@ julia> last([1; 2; 3; 4])
 4
 ```
 """
-last(a) = a[end]
+last(a) = first(Iterators.reverse(a))
+last(a::AbstractVector) = a[end]
```

Comment:
I think this can be for `AbstractArray`.

Reply:
I did that above, but @simeonschaub mentioned that since there are abstract arrays that don't support linear indexing, this might be the better default. Can give `AbstractArray` a try though and see how it goes

Reply:
IIRC, all arrays support linear indexing, but it just isn't required to be the most efficient index

Reply:
An array can also have a `lastindex` that isn't necessarily an integer.

Reply:
I thought so too, but `lastindex` is actually defined to always return an integer.